content/chainguard/chainguard-images/chainguard-registry/pull-through-guides/artifactory/artifactory-packages-pull-through/index.md

```diff
@@ -17,243 +17,233 @@ toc: true
+You **do not** need to set any values for the **User Name** or **Password / Access Token** fields, as the public repositories do not require authentication. Keep all the remaining fields set to their default values and click the **Create Remote Repository** button. This creates the remote repository and return you to the **Repositories** page. 
 
-<center><img src="ar-apk-5.png" alt="Screenshot showing a portion of the 'New Virtual Repository' page. The 'cg-wolfi' and 'cg-extras' repositories have been checked and moved to the 'Selected Repositories' column." style="width:750px;"></center>
-<br />
+After creating both repositories, generate and retrieve a token for each one as you did for the `cg-private` remote repository:
 
-Finally, click the **Create Virtual Repository** button. As before, a modal window will appear letting you know that the repository was created successfully. Click the **Set Up Generic Client** button at the bottom of this window to retrieve the token you'll need to test whether you can pull images through this repository. 
+1. On the Repositories page, find both remote repositories you just created.
+2. Click the ellipsis (**⋯**) all the way to the right of their respective rows.
+3. Select **Set me up**. 
+4. Click the **Generate Token & Create Instructions** button.
+5. Copy or take note of the tokens generated for your repositories.
 
-With this token in hand, create another environment variable:
+Next, copy the resulting tokens to create a pair of environment variables. For the `cg-chainguard` repository's token:
 
 ```shell
-export VIRTUAL_TOKEN=my-virtual-repo-token
+export CG_TOKEN=<cg-chainguard-token>
 ```
 
-You'll use this token to create one more Dockerfile in the next section.
+Then for the `cg-extras` repo's token:
 
-### Testing pull through with a virtual package repository
+```shell
+export EXTRAS_TOKEN=<cg-extras-token>
+```
 
-As outlined previously, create a Dockerfile that configures an image to use the virtual repository you just created. You can test this out however you like, but we will stick to one example here showing how to install a package into the built image:
+Following that, you can use these environment variables, along with a few created earlier in the guide, to create a Dockerfile as you previously did for the `cg-private` repository:
 
 ```shell
-cat > Dockerfile.virtual <<EOF
+cat > Dockerfile.repos <<EOF
 FROM cgr.dev/chainguard/python:latest-dev
 USER root
 RUN mv /etc/apk/repositories /etc/apk/repositories.disabled
-RUN echo 'https://$ARTIFACTORY_USER_PROFILE:$VIRTUAL_TOKEN@$ARTIFACTORY_HOSTNAME.jfrog.io/artifactory/cg-apk-virt/' >> /etc/apk/repositories
-RUN apk add nodejs-18
-USER nonroot
+RUN echo 'https://$ARTIFACTORY_USER_PROFILE:$CG_TOKEN@$ARTIFACTORY_HOSTNAME.jfrog.io/artifactory/cg-chainguard/' >> /etc/apk/repositories
+RUN echo 'https://$ARTIFACTORY_USER_PROFILE:$EXTRAS_TOKEN@$ARTIFACTORY_HOSTNAME.jfrog.io/artifactory/cg-extras/' >> /etc/apk/repositories
```

Comment:
I think it would actually be best to use three commands. First one to add public and overwrite the file with > 

and then extras with >> so its appended
and then maybe even again >> for private repo